main.go

```diff
@@ -1224,14 +1228,139 @@ func populatePackages(currentPackage string, onlyUpdates bool, onlyLatest bool,
 		if onlyUpdates && !updated {
 			continue
 		}
+
 		packageList = append(packageList, packageWrapper)
 	}
 
 	return packageList, nil
 }
 
-// func generateChanges(genpatch bool, save bool, commit bool, onlyUpdates bool, print bool) {
-func generateChanges(auto bool, stage bool) {
+// downloadIcons should only be used in a local machine by manual execution.
+// It will download all icons that contain URLs from the index.yaml file, if it is already downloaded it will keep it.
+// All downloaded icons will be saved in the assets/logos directory and for all of them will be created a upstream.yaml file in the respective packages folder.
+// The upstream.yaml file will contain the icon field under ChartMetadata to be overriden in the index.yaml file later by the overrideIcons function.
+func downloadIcons(c *cli.Context) {
+	currentPackage := os.Getenv(packageEnvVariable)
+	icons.CheckFilesStructure() // stop execution if file structure is not correct
+
+	packageList, err := populatePackages(currentPackage, false, false, false)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	// Convert packageList to PackageIconMap
+	var entriesPathsAndIconsMap icons.PackageIconMap = make(icons.PackageIconMap)
+	for _, pkg := range packageList {
+		entriesPathsAndIconsMap[pkg.Name] = icons.PackageIconOverride{
+			Name: pkg.Name,
+			Path: pkg.Path,
+			Icon: pkg.LatestStored.Metadata.Icon,
+		}
+	}
+
+	// Download all logos or retrieve the ones already downloaded
+	downloadedLogos := icons.DownloadFiles(entriesPathsAndIconsMap)
```

Comment:
I agree

Reply:
Fixed, the only point is that I prefer to use `icons` in the plural form because we will be working most of the time with several icons instead of a single icon.

Reply:
I see that occurrences of `logo` are mostly switched to `icon`, but there is still `assets/logos` - we should probably change this to `assets/icons`. There are a couple of other occurences as well. I used the command `git grep -r 'logo'` to find them.

Reply:
I am not sure but older versions of rancher might search specifically in `assets/logos`. 
Also if we change here, we should change this on  `rancher/charts` as well. 

So to be honest, since `logo` and `icon` seem like synonyms to me, I don't see the necessity of matching everything.

Reply:
> I am not sure but older versions of rancher might search specifically in assets/logos.

If so, we should understand how this works completely. Though I went back as far as rancher v2.6 and couldn't find (in a quick search) any references to `assets/logos`...?

I think that using consistent wording throughout `partner-charts-ci` will help us in the future, but if you feel strongly about this I won't argue :smile:

Reply:
I don't want to change this on `rancher/charts` now because I don't remember how it is structured there. 
This part is handled by `charts-build-script` in `rancher/charts`. 

Let's do like this, I will change every `icon` subjective to `logo` then.